### DISCUSSION TITLE
Avaloni UI app for kiosk mode in WIndows 11

### DISCUSSION THREAD

**Discussion by VladEfanovNewton** [AvaloniaUI/Avalonia#12976] [Q&A]
Hello,

I have developed a GUI app using Avalonia UI that, in some cases, launches another app through a specific interface and interacts with a local database.

I want to configure a user account on Windows 11 in such a way that the user only has access to the GUI app. I have learned about configuring kiosk mode on Windows, but I couldn't find my app in the list of available apps for kiosk mode.

I also came across the Shell Launcher, which allows configuring kiosk mode with third-party applications. However, I haven't found a comprehensive tutorial with examples.

Does anyone know the best way to achieve what I need? I would greatly appreciate it if you could provide an example of how to accomplish this.

**Comment by maxkatz6** [comment]
IIRC Kiosk Mode initially supported only UWP apps.
Most likely, it is possible now for win32 apps packaged with MSIX.
You can try to follow this guide https://learn.microsoft.com/en-us/windows/msix/desktop/desktop-to-uwp-packaging-dot-net
After that, it's possible to publish the packaging project as an MSIX application, which can be installed and most likely run in Kiosk mode.

I haven't tried it, but that's my first idea.

Alternatively, it looks like multi-app kiosk mode supports win32 apps, where you can specify only a single win32 app as well:
https://learn.microsoft.com/en-us/windows/configuration/lock-down-windows-11-to-specific-apps

**Comment by Yusuke1998** [comment]
Hello, did you manage to do it?

**Comment by ymebrugts** [comment]
I'm also interested to find out what's the best option here to get a single Avalonia app into official Windows Kiosk mode.
